com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -249,39 +247,94 @@ private void OnValidate()
                 };
             }
 
+            //If true, check to see if there is a defined player prefab in the Network prefab list
+            //This helps migrate v0.1.0 projects (i.e. no PlayerPrefab was assigned) as well as helps detect
+            //specific interface related issues related to the NetworkPrefabs list.
+            var scanForPlayerPrefab = NetworkConfig.PlayerPrefab == null;
+
+            //During OnValidate we will always clear out NetworkPrefabOverrideLinks and rebuild it in order to
+            //detect specific edge case scenarios (See NtworkConfig.PreviousPlayerPrefab comments)
+            NetworkConfig.NetworkPrefabOverrideLinks.Clear();
+
+            //Check network prefabs and assign to dictionary for quick look up
             for (int i = 0; i < NetworkConfig.NetworkPrefabs.Count; i++)
             {
                 if (NetworkConfig.NetworkPrefabs[i] != null && NetworkConfig.NetworkPrefabs[i].Prefab != null)
                 {
-                    if (NetworkConfig.NetworkPrefabs[i].Prefab.GetComponent<NetworkObject>() == null)
+                    var networkObject = NetworkConfig.NetworkPrefabs[i].Prefab.GetComponent<NetworkObject>();
+                    if (networkObject == null)
                     {
                         if (NetworkLog.CurrentLogLevel <= LogLevel.Normal)
                         {
                             NetworkLog.LogWarning($"{nameof(NetworkPrefab)} [{i}] does not have a {nameof(NetworkObject)} component");
                         }
                     }
-                }
-            }
+                    else
+                    {
+                        //If someone is transitioning to this new format, then go ahead and populate the PlayerPrefab reference for them.
+                        if (scanForPlayerPrefab && NetworkConfig.NetworkPrefabs[i].IsPlayer)
+                        {
+                            //If we find a NetworkPrefab where IsPlayer is true and we have
+                            //no PlayerPrefab nor did we have one previously (i.e. we didn't just delete it),
+                            //then we want to set the default player prefab.  We also want to set the
+                            //previous player prefab value in the event the user decides to delete the
+                            //PlayerPrefab value from within the editor inspector view.
+                            //(See NtworkConfig.PreviousPlayerPrefab comments for more information)
+                            if (NetworkConfig.PreviousPlayerPrefab == null)
```

Comment:
I guess I'm kind of on the fence.  I like the defensive programming against the old scheme, but I also would like to avoid maintaining the work around code nor have to remember to eventually get rid of it.  

Could we just detect the old scheme is in use, then signal the user that they need to re-configure their network prefabs?

Reply:
Until we implement something like [custom serialization](https://docs.unity3d.com/2020.2/Documentation/Manual/script-Serialization-Custom.html) with our serialized NetworkConfig class and provide some form of version number to check against (i.e. determine what the version is and then "upgrade" or "downgrade" the configuration), we will have to either break existing configurations or handle things like the above.

Until that happens, we just need to determine what we want to accomplish.  
**Side Note:**  _We can easily keep track of this by just creating a new Jira story that references the areas that need to be addressed and then assign it to the next milestone after v1.0.0.   That way we won't have to remember to remove it as we will already have a task to do so in the not too distant future._

Reply:
we're already breaking the old NetworkPrefab setup UI and I think it's safe to change API (therefore already serialized/saved assets) this time here in v1.0.0-pre release. we should address this case by case basis. I'll bring this up in our internal discussions.